### DISCUSSION TITLE
Changes of TCP

### DISCUSSION THREAD

**Discussion by wienans** [UniversalRobots/Universal_Robots_ROS2_Driver#945] [Q&A] [answered]
Can the driver change TCP to a other preconfigured one. Thinking about a pick and place situation. Loaded and unloaded gripper require different Center of mass and so on. If not how is that supposed to be done?

**Comment by RobertWilbrandt** [comment, marked as answer]
This feature is supported. Take a look at the `io_and_status_controller`s `set_payload` service. You can use it like this:
```
ros2 service call /io_and_status_controller/set_payload ur_msgs/srv/SetPayload "{center_of_gravity: {z: 0.2}, mass: 3.5}"
```
Does this work for you?

**Reply by wienans** [reply]
Sorry my bad. Was so focused on TCP so didn't got the bridge between Payload and TCP. And i looks like that this interfaces are not documented in the Docs at all https://docs.ros.org/en/ros2_packages/rolling/api/ur_robot_driver/ROS_INTERFACE.html

So the process would be to teach the TCP in the Robot and then get that values to set them via ROS. I assume it is not supported to directly switch to the configured Robot TCP as this is more general. Only downside is that if i want to use the teaching of the Robot need to put the values twice, so values can get messed up / forgot to be changed.
